osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -174,6 +174,22 @@ protected override void PreImport(BeatmapSetInfo beatmapSet)
             if (beatmapSet.Beatmaps.Any(b => b.BaseDifficulty == null))
                 throw new InvalidOperationException($"Cannot import {nameof(BeatmapInfo)} with null {nameof(BeatmapInfo.BaseDifficulty)}.");
 
+            var dbContext = ContextFactory.Get();
+
+            // Workaround System.InvalidOperationException
```

Comment:
does this need to be copy pasted across multiple places? I see no reason not to hide this re-query operation under a helper or extension method.

Reply:
Yep, I think if this is isolated to an extension method it should clean up quite nicely.